airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -88,6 +95,29 @@ def wrapper(*args, **kwargs) -> T:
     return cast(T, wrapper)
 
 
+def provide_bucket_name_async(func: T) -> T:
```

Comment:
I dont see any unit tests for all the many new methods you created. Could you please add them?

Reply:
@vincbeck added tests here: https://github.com/apache/airflow/pull/31018/commits/0785452f46d7653abc70281b33414cd3a85c40b2